src/containers/permissionssummary/PermissionsSummaryReducer.js

```diff
@@ -56,16 +55,21 @@ export default function reducer(state :Immutable.Map<*, *> = INITIAL_STATE, acti
             }
           }
         };
-        return state.mergeDeep(rolePermissionsMerge);
+
+        const rolePermissionsMergeImmutable = Immutable.fromJS(rolePermissionsMerge);
+        const mergedState = state.mergeDeep(rolePermissionsMergeImmutable);
```

Comment:
`mergeDeep` has caused problems in the past, and is probably not what we want to do. What are you trying to do here?

Reply:
Ya mergeDeep was the main issue. Took a while to track down what exactly was happening. userPermissions was overwriting rolePermissions. It seems that if an object has a mix of regular + immutable values, mergeDeep can get screwy. I'm trying to save permissions in a nested state object, so mergeDeep is pretty much required I think. I did this instead of a flat structure because at the time it seemed like a more optimal way for components to load them.

Reply:
Also it seems that fromJS/toJS also might not always fully convert an object with mixed regular/immutable values. This seemed to be part of the problem too. FYI.